### PR TITLE
add pre-push ci checks

### PR DESCRIPTION
**Pull request by npmaile**
this is to save time in the pull request -> ci failure -> fix problems  workflow
It will be a little embarrassing if this doesn't pass CI.

### CODE REVIEW

File: beforepush.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#setting up environment for terraform operations
+export VAULT_ADDR="http://FakeVaultAddr:8200"
+export VAULT_TOKEN="FakeVaultToken"
+export VAULT_DEV_ROOT_TOKEN_ID="FakeDevRootTokenId"
+
+tfdirs='static_ips
+dns
+spinnaker
+halyard'
```
Comment (devorbitus) [inline beforepush.sh]: everyone forgets all about poor old `monitoring-alerting`, it's as if she doesn't even matter when in fact she matters very much 😢

File: beforepush.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+for file in $files; do
+	echo "checking $file"
+	if ! shellcheck "$file"; then
+		echo "shellcheck failed for $file"
+		exit 1
+	fi
+done
+
+#setting up environment for terraform operations
+export VAULT_ADDR="http://FakeVaultAddr:8200"
```
Comment (devorbitus) [inline beforepush.sh]: I would suggest only setting these if they are not already set

File: beforepush.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+if ! command -v shellcheck; then
+	echo "shellcheck not found on this machine. Please install from https://www.shellcheck.net/"
```
Comment (devorbitus) [inline beforepush.sh]: You might even be more specific and give them this address too `https://github.com/koalaman/shellcheck#installing`

File: beforepush.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+tfdirs='static_ips
+dns
+spinnaker
+halyard'
+
+for tfdir in $tfdirs; do
+	cd "$tfdir" || { echo "failed to cd to $tfdir"; exit 1; }
+	echo "checking terraform scripts in $tfdir"
+	if ! terraform init -backend=false .; then
```
Comment (devorbitus) [inline beforepush.sh]: won't this overwrite the `.terraform` directory that I would have if I had been working on terraform?

File: beforepush.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		echo "terraform init failed for $tfdir"
+		exit 1
+	fi
+	if ! terraform validate; then
+		echo "terraform validate failed for $tfdir"
+		exit 1
+	fi
+	cd ..
```
Comment (devorbitus) [inline beforepush.sh]: At this point, won't I now have a `.terraform` directory that won't be usable for continued development without re-running `terraform init`?

File: beforepush.sh (hunk 1 above)
Comment (npmaile) [reply beforepush.sh]: whoops.

Comment (npmaile) [reply beforepush.sh]: added

File: beforepush.sh (hunk 2 above)
Comment (npmaile) [reply beforepush.sh]: moved to docker container where they are guaranteed not set.

**Comment by npmaile** [comment]
everything is docker now. Should be good to go.

**Pull request merged**